html_text/html_text.py

```diff
@@ -44,30 +47,89 @@ def parse_html(html):
 _whitespace = re.compile(r'\s+')
 _has_trailing_whitespace = re.compile(r'\s$').search
 _has_punct_after = re.compile(r'^[,:;.!?"\)]').search
-_has_punct_before = re.compile(r'\($').search
+_has_open_bracket_before = re.compile(r'\($').search
+
 
+def html_to_text(tree, guess_punct_space=True, guess_page_layout=False):
```

Comment:
I think we should keep this function private - rename it to `_html_to_text`, remove from `__init__` exports. The API becomes rather confusing: there is `html_text.extract_text` and `html_text.html_to_text`, which do almost the same - `extract_text` supports html as a string in addition to lxml trees, and also does cleaning on its own; `html_to_text` only works on lxml trees, and is not doing cleaning. 

If allowing to pass an already cleaned tree is important, we can add an argument to `extract_text` - though this can be done later.

Reply:
@Kebniss right, though let's not worry about adding `clean=True` argument now. Main use cases are 

* user has an already cleaned tree, and wants to save some time by avoiding cleaning again;
* user wants to change cleaning options.